contracts/token/Token.sol

```diff
@@ -132,23 +123,36 @@ contract Token is IERC20 {
     /// @param spender must have allowance for the caller of at least `subtractedValue`. It cannot be the zero address.
     /// @param subtractedValue The subtracted value.
     /// @return Returns a boolean value indicating whether the operation succeeded.
-    function decreaseAllowance(
-        address spender,
-        uint256 subtractedValue
-    ) public returns (bool) {
+    function decreaseAllowance(address spender, uint256 subtractedValue) public returns (bool) {
         address owner = msg.sender;
         uint256 currentAllowance = _allowances[owner][spender];
-        require(
-            currentAllowance >= subtractedValue,
-            "ERC20: decreased allowance below zero"
-        );
+        require(currentAllowance >= subtractedValue, "ERC20: decreased allowance below zero");
         unchecked {
             _approve(owner, spender, currentAllowance - subtractedValue);
         }
 
         return true;
     }
 
+    /// @notice Moves `amount` tokens from the caller's account to `dest`, which is AccountId32 type account.
+    /// @dev It'll emit an {TransferToAccountId32} event. The caller must have a balance of at least `amount`.
+    /// @param dest The dest AccountId32 type account, it cannot be the zero AccountId32.
+    /// @param amount The transfer amount.
+    /// @return Returns a boolean value indicating whether the operation succeeded.
+    function transferToAccountId32(bytes32 dest, uint256 amount) public returns (bool) {
+        address from = msg.sender;
+        require(from != address(0), "ERC20: transfer from the zero address");
```

Comment:
no need to check this and if `msg.sender` is indeed all zero address, that must be triggered by some root calls